astro/src/content/docs/get-started/start-here/step-5.mdx

```diff
@@ -0,0 +1,101 @@
+
+When using FusionAuth as an OAuth identity provider, user's are logged in using what is known as the Authorization Code Grant. This leverages FusionAuth's OAuth frontend to authenticate users and provide an access token to the application.
+
+Access tokens are designed to have short lifespans though. In most cases, an access token will only be valid for a few minutes before it expires. It would be brutal to force users to log in every few minutes. Instead, it would be ideal to control how log the user is logged in via a user session.
+
+Luckily, the OAuth standard provides a simple way to manage user sessions. This is done using refresh tokens.
+
+Refresh tokens are long lived tokens that should never be shared with any other application, user, or system. These are the most secure tokens and therefore most be controlled. In many cases, refresh tokens are stored in cookies in the browser. These cookies must be marked as `HttpOnly` and `Secure`. These two settings ensure that the refresh token cannot be stolen in any way. Ideally, the access token is also marked with the same attributes.
+
+Our example app handles refresh tokens automatically. Let's look at how this is accomplished.
+
+Open the file `src/sdk.ts` and locate the function named `logInUser`. This function is where the access token, refresh token, and id token (part of the OpenID Connect specification) are written out to the browser as cookies. This function looks like this:
+
+```typescript
+logInUser(accessToken: AccessToken, res: Response) {
+  res.cookie(this.configuration.accessTokenCookieName, accessToken.access_token, { httpOnly: true });
+  res.cookie(this.configuration.idTokenCookieName, JSON.stringify(jose.decodeJwt(accessToken.id_token)), { httpOnly: false });
+
+  if (this.configuration.enableRefreshTokens && accessToken.refresh_token) {
+    res.cookie(this.configuration.refreshTokenCookieName, accessToken.refresh_token, { httpOnly: true });
+  }
+}
+```
+
+You can see that the access and refresh token are written out as `HttpOnly`. Our example application doesn't run over TLS, therefore we can't set the `Secure` flag. But for production applications, you should use TLS and mark cookies as `Secure`.
+
+Next, let's look at how the refresh token is used when the access token expires. Scroll to the `getUser` function in this file. This function loads the same cookie that was written out above. We are using the `jose` library, which throws an exception when the access token has expired. This exception is handled in the `catch` block like this:
+
+```typescript
+try {
+  payload = (await jose.jwtVerify(accessToken, this.JWKS, {
+    issuer: this.configuration.oauthIssuer,
+    audience: this.configuration.applicationId,
+  })).payload;
+} catch (e) {
+  if (e instanceof jose.errors.JWTExpired) {
+    // ...
+  }
+}
+```
+
+Inside this `catch` block, we check if the exception is due to an expired JWT. If the JWT is expired, we refresh it using this code:
+
+```typescript
```

Comment:
If I'm not reading carefully, it's easy to miss that this block of code is what's in the `catch` block. It might be clearer to move this code into a `attemptTokenRefresh` function, and call that from the try block and put a call to it in the `catch` block and not just a "// ...". But if you did that then you'd probably want to move the cookie-writing code into a function so that the `catch` block doesn't have a ton of code in it in earlier places in the guide, where it's just going to be confusing/distracting.